pkgs/top-level/all-packages.nix

```diff
@@ -290,7 +291,7 @@ in
         then { inherit rev fetchSubmodules; url = "${baseUrl}.git"; }
         else ({ url = "${baseUrl}/archive/${rev}.tar.gz"; } // privateAttrs)
       ) // passthruAttrs // { inherit name; };
-  in fetcher fetcherArgs // { meta.homepage = baseUrl; inherit rev; };
+  in fetcher fetcherArgs // { meta.homepage = baseUrl; inherit nakedUrl rev; };
```

Comment:
I disagree with adding `nakedUrl`. It's a composition of some input values, which just happens to be useful for Go packages. Instead, make the individual values available and do the composition elsewhere (`pkgs/development/go-modules/generic/default.nix`).

Reply:
Agreed. `go get` has some heuristic to translate goPackagePath to URLs, I don't know how uniform that is. It would be better to detect the src.url in buildGoPackage.